src/main/java/Duke.java

```diff
@@ -1,10 +1,323 @@
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        displayWelcomeMessage();
+        loadHistory();
+        inputAndExecuteCommand();
+        saveHistory();
+        displayExitMessage();
+    }
+
+    private static void displayWelcomeMessage() {
+        String logo = "         __    _    _              ____        __           \n"
+                +"        / /_  (_)  (_)___ ___     / __ \\__  __/ /_____      \n"
+                +"       / __ \\/ /  / / __ `__ \\   / / / / / / / //_/ _ \\     \n"
+                +"      / / / / /  / / / / / / /  / /_/ / /_/ / ,< /  __/     \n"
+                +"     /_/ /_/_/  /_/_/ /_/ /_/  /_____/\\__,_/_/|_|\\___/     \n";
+        System.out.print(logo + "\n");
+        System.out.print("What do you have to do today?\n");
+        System.out.print(COMMANDS + "\n");
+    }
+
+    private static void loadHistory() {
+        String home = System.getProperty("user.dir");
+        createDirectory(home);
+        loadDataFile(home);
+    }
+
+    private static void createDirectory(String home) {
+        Path path = Paths.get(home, "data");
+        try {
+            Files.createDirectory(path);
+        } catch (FileAlreadyExistsException e) {
+            // do nothing
+        } catch (Exception e) {
+            String errorMessage = "ERROR creating directory!\n" + e.getLocalizedMessage();
+            printWithBorder(errorMessage);
+        }
+    }
+
+    private static void loadDataFile(String home) {
+        Path path = Paths.get(home, "data", "duke.txt");
+        if (Files.notExists(path)) {
+            return;
+        }
+        try {
+            List<String> data = Files.readAllLines(path);
+            for (String line : data) {
+                loadTask(line);
+            }
+        } catch (Exception e) {
+            String errorMessage = "ERROR retrieving data!\n" + e.getLocalizedMessage();
+            printWithBorder(errorMessage);
+        }
+    }
+
+    private static void loadTask(String line) {
+        String[] tokens = line.split("~");
+        String taskType = tokens[0];
+        String isDone = tokens[1];
+        String description = tokens[2];
+        Task task = new Task(description);
+        switch (taskType) {
+        case "Todo":
+            task = new Todo(description);
+            break;
+        case "Deadline":
+            String by = tokens[3];
+            task = new Deadline(description, by);
+            break;
+        case "Event":
+            String at = tokens[3];
+            task = new Event(description, at);
+            break;
+        }
+        if (isDone == "1") {
```

Comment:
Here, `1` seems like a magic number. Could there be a better way of representing the values of `isDone`, while avoiding magic numbers?

Reply:
thank you! i will change "1" to the string version of true